### PR TITLE
Update url-parse typings to 1.4

### PR DESCRIPTION
**Pull request by DimitryDushkin**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [ ] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [ ] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [ ] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [ ] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

If changing an existing definition:
- [ ] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [ ] Increase the version number in the header if appropriate.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

If removing a declaration:
- [ ] If a package was never on DefinitelyTyped, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [ ] Delete the package's directory.
- [ ] Add it to `notNeededPackages.json`.

### CODE REVIEW

**Comment by DimitryDushkin** [comment]
Remove typings for using URL with out `new`, because it fails anyway and I didn't managed to make it work.

File: types/url-parse/index.d.ts
Comment (mhegazy) [inline types/url-parse/index.d.ts]: do not remove existing authors. just add your name to the list.

File: types/url-parse/url-parse-tests.ts
Comment (mhegazy) [inline types/url-parse/url-parse-tests.ts]: calling parse as a funciton seems to be allowed in the documentation. to do that the exported value should have a type like:

```ts
declare const parse: {
    new  (address: string, location?: string | object, parser?: boolean | QueryParser): URLParse;
    (address: string, location?: string | object, parser?: boolean | QueryParser): URLParse;
};
export = parse;
```

**Review by mhegazy — DISMISSED** [review]

**Comment by DimitryDushkin** [comment]
Thank you for review! Sending updates.

**Comment by DimitryDushkin** [comment]
So, can we merge?)

**Comment by RyanCavanaugh** [comment]
Good to merge once merge conflicts are fixed 👍

**Comment by DimitryDushkin** [comment]
Done

**Comment by harisiva** [comment]
This appears to be a breaking change from the previous version (1.1.1). Was this intentional?

The https://github.com/DefinitelyTyped/DefinitelyTyped/pull/27125 exported out the URL definition, but this PR reverted that change.

**Comment by SamRaymer** [comment]
Boolean arguments for the node implementation are no longer supported with this change too

**Comment by DimitryDushkin** [comment]
@harisiva @SamRaymer hey! Could you pls provide valid example which fails on updated typings?

> Boolean arguments for the node implementation are broken with this change too
Correct support of (from docs)
```
var parse = require('url-parse')
  , url = parse('https://github.com/foo/bar', true);
```
one of the goals of this PR, now it works.

**Comment by DimitryDushkin** [comment]
Yeah! I mentioned it too lately and sent PR — https://github.com/DefinitelyTyped/DefinitelyTyped/pull/27531

**Comment by harisiva** [comment]
@DimitryDushkin, the repro is in the test case that was overwritten by this PR.

https://github.com/DefinitelyTyped/DefinitelyTyped/pull/26947/files#diff-f80cf0a6fca8e5340a1fe7143d1f8e34L4

```ts
const url2: parse.URL = parse("https://github.com/foo/bar?baz=true");
```

The test case was enforcing that the URL type was exported out of the module and could be used by consumers of `url-parse`.

It appears that this PR blew undid all the changes that came through in #27125. 

A bad merge?

**Comment by DimitryDushkin** [comment]
> A bad merge?

I didn't really looked at #27125 cause it looked like in #26947 I've counted all cases anyway. Sorry for that.

**Pull request merged**
